plugins/inputs/huebridge/huebridge_test.go

```diff
@@ -0,0 +1,95 @@
+	require.Len(t, plugin.Bridges, 4)
+	require.NotEmpty(t, plugin.RemoteClientId)
+	require.NotEmpty(t, plugin.RemoteClientSecret)
+	require.NotEmpty(t, plugin.RemoteCallbackUrl)
+	require.NotEmpty(t, plugin.RemoteTokenDir)
+	require.NotNil(t, plugin.RoomAssignments)
+	require.Len(t, plugin.RoomAssignments, 2)
+	require.Equal(t, config.Duration(60*time.Second), plugin.Timeout)
+	require.Equal(t, "secret", plugin.TLSKeyPwd)
+	require.True(t, plugin.InsecureSkipVerify)
+}
+
+func TestInitBridges(t *testing.T) {
+	// Start mock server
+	bridgeMock := mock.Start()
+	require.NotNil(t, bridgeMock)
+	defer bridgeMock.Shutdown()
+	// Prepare token dir
+	tokenDir, err := os.MkdirTemp("", "TestInitBridges")
+	require.NoError(t, err)
+	tokenFile := filepath.Join(tokenDir, mock.MockClientId, strings.ToUpper(mock.MockBridgeId)+".json")
+	bridgeMock.WriteTokenFile(tokenFile)
+	// Actual test
+	plugin := &HueBridge{Timeout: config.Duration(10 * time.Second)}
+	plugin.Bridges = append(plugin.Bridges, fmt.Sprintf("address://%s:%s@%s/", mock.MockBridgeId, mock.MockBridgeUsername, bridgeMock.Server().Host))
+	plugin.Bridges = append(plugin.Bridges, fmt.Sprintf("cloud://%s:%s@%s/discovery", mock.MockBridgeId, mock.MockBridgeUsername, bridgeMock.Server().Host))
+	plugin.Bridges = append(plugin.Bridges, fmt.Sprintf("mdns://%s:%s@/", mock.MockBridgeId, mock.MockBridgeUsername))
+	plugin.Bridges = append(plugin.Bridges, fmt.Sprintf("remote://%s:%s@%s/", mock.MockBridgeId, mock.MockBridgeUsername, bridgeMock.Server().Host))
+	plugin.RemoteClientId = mock.MockClientId
+	plugin.RemoteClientSecret = mock.MockClientSecret
+	plugin.RemoteTokenDir = tokenDir
+	plugin.InsecureSkipVerify = true
+	plugin.Log = testutil.Logger{Name: "huebridge"}
+	err = plugin.Init()
+	require.NoError(t, err)
+}
+
+func TestGatherLocal(t *testing.T) {
+	// Start mock server
+	bridgeMock := mock.Start()
+	require.NotNil(t, bridgeMock)
+	defer bridgeMock.Shutdown()
+	// Actual test
+	plugin := &HueBridge{Timeout: config.Duration(10 * time.Second)}
+	plugin.Bridges = append(plugin.Bridges, fmt.Sprintf("address://%s:%s@%s/", mock.MockBridgeId, mock.MockBridgeUsername, bridgeMock.Server().Host))
+	plugin.RoomAssignments = map[string]string{"Name#7": "Name#15"}
+	plugin.Log = testutil.Logger{Name: "huebridge"}
+	err := plugin.Init()
+	require.NoError(t, err)
+	acc := &testutil.Accumulator{}
+	err = acc.GatherError(plugin.Gather)
+	require.NoError(t, err)
+	testMetric(t, acc, "testdata/metrics/huebridge.txt", telegraf.Gauge)
+}
+
+func testMetric(t *testing.T, acc *testutil.Accumulator, file string, vt telegraf.ValueType) {
+	parser := &influx.Parser{}
+	err := parser.Init()
+	require.NoError(t, err)
+	expectedMetrics, err := testutil.ParseMetricsFromFile(file, parser)
+	for index := range expectedMetrics {
+		expectedMetrics[index].SetType(vt)
+	}
+	require.NoError(t, err)
+	testutil.RequireMetricsEqual(t, expectedMetrics, acc.GetTelegrafMetrics(), testutil.IgnoreTime(), testutil.SortMetrics())
+}
```

Comment:
Please merge the two. Furthermore, you could directly define the expected metrics here and get rid of the file...

Reply:
Merged it partly. I have kept the test data file as well the loading code separated to keep the focus on the actual test logic. The actual testing is now done within the test function.